app/controllers/patients_controller.rb

```diff
@@ -93,7 +98,31 @@ def find_patient
     @patient = Patient.find params[:id]
   end
 
+  def format_dates(params)
+    new_params = params
+    if !new_params[:appointment_date].blank?
+      new_params[:appointment_date] = Date.strptime(params[:appointment_date], '%m-%d-%Y')
+    end
+    if !(new_params[:fulfillment] && new_params[:fulfillment][:procedure_date]).nil?
+      new_params[:fulfillment][:procedure_date] = Date.strptime(params[:fulfillment][:procedure_date], '%m-%d-%Y')
+    end
+
+    if !new_params[:initial_call_date].blank?
+      begin
+        new_params[:initial_call_date] = Date.strptime(params[:initial_call_date], '%m-%d-%Y')
+      rescue
+        # for patient_controller_test where date parameters include timezones.
+        # culprit is in test/factories/patient.rb
+        new_params[:initial_call_date] = Date.strptime(params[:initial_call_date], '%Y-%m-%d %H:%M:%S %z')
+      end
+    end
+
+    return new_params
+  end
+
   # Strong params divided up by partial
+  #TODO how to unfreeze part of array, i.e appointment date, procedure_date, check date
```

Comment:
I had to look this up for unrelated reasons today -- when an array freezes, it swaps to an immutable state, so you're either stuck with it or you have to redeclare it.